shipyard-controller/handler/servicemanager.go

```diff
@@ -39,6 +45,11 @@ func (sm *serviceManager) createService(projectName string, params *operations.C
 	}
 
 	for _, stage := range stages {
+		sm.logger.Info(fmt.Sprintf("Validating service %s", *params.ServiceName))
+		if err := validateServiceName(projectName, stage.StageName, *params.ServiceName); err != nil {
+			return sm.logAndReturnError(fmt.Sprintf("could not create service %s in stage %s of project %s: %s", *params.ServiceName, stage.StageName, projectName, err.Error()))
+		}
+
```

Comment:
The validation itself looks okay, but I think it's in the wrong place.

In the worst case you will create services in two stages (e.g., `dev`, `staging`), but in the third stage (`production`) you hit the length and it will return.

I know the for-loop is not optimal in itself, but to avoid introducing new side-effects, let's move your check in a separate for loop above the existing once.

```
	for _, stage := range stages {
		sm.logger.Info(fmt.Sprintf("Validating service %s", *params.ServiceName))
		if err := validateServiceName(projectName, stage.StageName, *params.ServiceName); err != nil {
			return sm.logAndReturnError(fmt.Sprintf("could not create service %s in stage %s of project %s: %s", *params.ServiceName, stage.StageName, projectName, err.Error()))
		}
	}

	for _, stage := range stages {
		...
	}
```

Reply:
@christian-kreuzberger-dtx thanks that makes sense, I thought on that but was unsure. thanks again for your review
PTAL